addons/udes_stock_picking_batch/models/stock_move_line.py

```diff
@@ -0,0 +1,128 @@
+from collections import defaultdict
+from datetime import datetime
+import logging
+
+from odoo import models, fields, api, _
+from odoo.tools.float_utils import float_compare, float_round
+from odoo.osv import expression
+from odoo.exceptions import ValidationError
+
+_logger = logging.getLogger(__name__)
+
+
+class StockMoveLine(models.Model):
+    _inherit = "stock.move.line"
+
+    def _determine_priority_skipped_moveline(
+        self, skipped_product_ids=None, skipped_move_line_ids=None
+    ):
+        """Returns a priority move line based on the first move line found
+        that matches either the skipped product ids or skipped move_line_ids.
+        """
+        if not self:
+            return False
+
+        priority_mls = False
+        # Provided skipped lists are ordered, first matching move line will have priority
+        if skipped_product_ids:
+            for skipped_prod_id in skipped_product_ids:
+                priority_mls = self.filtered(lambda ml: ml.product_id.id == skipped_prod_id)
+                if priority_mls:
+                    break
+        if skipped_move_line_ids:
+            for skipped_ml in skipped_move_line_ids:
+                priority_mls = self.filtered(lambda ml: ml.id == skipped_ml)
+                if priority_mls:
+                    break
+        # Check if we found a priority move lines
+        if priority_mls:
+            return priority_mls[0]
+        else:
+            return False
+
+    def next_task_sort(self):
+        """Creating a method so we can override it by sorting another combination.
+        Would be easier and meaningful than to extend sort_by_key as we may need to change the
+        behaviour of next_task_sort result rather than changing the behaviour of sort_by_key"""
+        return self.sort_by_key()
+
+    def _prepare_task_info(self):
```

Comment:
Do we have a definition somewhere of what a task is? I know this might be out of scope but if we don't have any explanations of what a task is, can we add one? Maybe in the readme? I can see lots of functions that talk about a task but no explanations anywhere of what a task is and how it is relevant to a batch.

Reply:
@armandcela Will resolve and suggest we look into this later as part of another story as you are time bounded right now.